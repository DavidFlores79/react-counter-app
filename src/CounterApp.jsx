import React, { useState } from 'react'
import { FirstApp } from './components/FirstApp'
import PropTypes from 'prop-types';




export const CounterApp = ({ value }) => {

    const [ counter, setCounter ] = useState(value);

    const increment = () => {
        console.log('increment');
        setCounter( counter + 1 )
    }

    const decrement = () => {
        console.log('decrement');
        setCounter( counter - 1 )
    }

    const reset = () => {
        console.log('reset');
        setCounter( value )
    }


    return (
        <div>
            <h1>CounterApp</h1>
            <p>Soy un subtitulo</p>
            <p>subtitulo2</p>
            <p>subtitulo2</p>
            <h2>{counter}</h2>
            <button onClick={increment}>+1</button>
            <button onClick={decrement}>-1</button>
            <button aria-label='btn-reset' onClick={reset}>Reset</button>
            {/* <FirstApp name="David" edad={ 44 } /> */}
        </div>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
