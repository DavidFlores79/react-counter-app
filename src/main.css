html, body {
    background-color: #21232A;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 20px;
    color: #FFFFFF;
}

button {
    padding: 10px 25px;
    font-size: 1.3rem;
    margin: 5px;
}