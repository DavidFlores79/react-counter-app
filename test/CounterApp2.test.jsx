import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Pruebas refactorizadas en CounterAPP', () => {

    const initialValue = 15;

    test('debe hacer match con el snapshot', () => {

        const { container } = render(<CounterApp value={initialValue} />)
        expect(container).toMatchSnapshot();
    })

    test('debe mostrar valor unicial de 100', () => {

        render(<CounterApp value={100} />)
        expect(screen.getByText(100)).toBeTruthy()
    })

    test('debe mostrar un h2', () => {

        render(<CounterApp value={100} />)
        expect(screen.getByRole('heading', { level: 2 }).innerHTML).toBe('100')
    })

    test('debe incrementar valor con boton +1', () => {

        render(<CounterApp value={initialValue} />)
        fireEvent.click(screen.getByText('+1'))
        expect(screen.getByText('16')).toBeTruthy();
    })

    test('debe decrementar valor con boton -1', () => {

        render(<CounterApp value={initialValue} />)
        fireEvent.click(screen.getByText('-1'))
        expect(screen.getByText('14')).toBeTruthy();
        fireEvent.click(screen.getByText('-1'))
        expect(screen.getByText('13')).toBeTruthy();
    })

    test('debe funcionar el boton reset', () => {

        render(<CounterApp value={initialValue} />)
        fireEvent.click(screen.getByText('+1'))
        fireEvent.click(screen.getByText('-1'))
        fireEvent.click(screen.getByText('-1'))
        fireEvent.click(screen.getByText('Reset'))
        expect(screen.getByText(initialValue)).toBeTruthy();
    })
    
    test('debe funcionar el boton reset aria-label', () => {

        render(<CounterApp value={initialValue} />)
        fireEvent.click(screen.getByText('+1'))
        fireEvent.click(screen.getByText('-1'))
        fireEvent.click(screen.getByText('-1'))
        fireEvent.click(
            screen.getByRole('button', { name: 'btn-reset'})
        )
        expect(screen.getByText(initialValue)).toBeTruthy();
    })




})