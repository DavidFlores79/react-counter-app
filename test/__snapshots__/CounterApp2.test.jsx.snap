// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas refactorizadas en CounterAPP debe hacer match con el snapshot 1`] = `
<div>
  <div>
    <h1>
      CounterApp
    </h1>
    <p>
      Soy un subtitulo
    </p>
    <p>
      subtitulo2
    </p>
    <p>
      subtitulo2
    </p>
    <h2>
      15
    </h2>
    <button>
      +1
    </button>
    <button>
      -1
    </button>
    <button
      aria-label="btn-reset"
    >
      Reset
    </button>
  </div>
</div>
`;
